src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -14,6 +14,7 @@ namespace System.IO
     internal sealed partial class UnixFileSystem : FileSystem
     {
         internal const int DefaultBufferSize = 4096;
+        private static char[] s_EscapeCharacters = new char[] { '\\', '[' };
```

Comment:
If we keep this, it should be readonly and `s_escapeCharacters` instead of `s_EscapeCharacters`, but I question whether the IndexOfAny is actually helping.